src/history/history-detail-overview.component.tsx

```diff
@@ -39,49 +39,46 @@ const HistoryDetailOverview: React.FC<HistoryDetailOverviewProps> = () => {
   const [ips, setIps] = useState({
     history: null,
     isLoading: true,
-    error: "",
-  })
-  
+    error: '',
+  });
+
   useEffect(() => {
     if (history != null) {
-      setIps((ps) => ({...ps, history: history, isLoading: isLoading, error: error?.message}));
+      setIps((ps) => ({ ...ps, history: history, isLoading: isLoading, error: error?.message }));
     }
-  },[error, history, isLoading])
-
+  }, [error, history, isLoading]);
 
   const handleSubmit = () => {
     setIsSubmitting(true);
-    createIpsResource(
-      uuid, 
-      abortController,
-    ).then((response) => {
-      if (response.status === 200) {
-        setIps((ps) => ({...ps, history: history, isLoading: isLoading, error: error?.message}));
+    createIpsResource(uuid, abortController)
+      .then((response) => {
+        if (response.status === 200) {
+          setIps((ps) => ({ ...ps, history: history, isLoading: isLoading, error: error?.message }));
+          setIsSubmitting(false);
+          showSnackbar({
+            isLowContrast: true,
+            kind: 'success',
+            title: t('ipsCreated', 'IPS'),
+            subtitle: t('ipsNowAvailable', 'The IPS has been updated and is now visible in the Patient History.'),
+          });
+        }
+      })
+      .catch((err) => {
+        setIps((ps) => ({ ...ps, history: null, isLoading: false, error: err?.message || 'Failed to fetch IPS' }));
         setIsSubmitting(false);
         showSnackbar({
-          isLowContrast: true,
-          kind: 'success',
-          title: t('ipsCreated', 'IPS'),
+          title: t('ipsCreationError', 'IPS'),
+          kind: 'error',
+          isLowContrast: false,
           subtitle: t(
-            'ipsNowAvailable',
-            'The IPS has been updated and is now visible in the Patient History.',
+            'checkForServerAvailability',
+            'The Fhir server maybe unreachable or the IPS generation process exited with an error!',
```

Comment:
FHIR like IPS is an acronym (Fast Healthcare Interoperability Resources)

```suggestion
            'The FHIR server maybe unreachable or the IPS generation process exited with an error!',
```

Reply:
It might be better to pass through a more detailed error from the backend. At the very least, the message should be logged to the console as an error.